src/transformStateWithClones.js

```diff
@@ -8,6 +8,33 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const stateHistory = [];
+  let previousState = { ...state };
+
+  for (const action of actions) {
+    const { type } = action;
+
+    let newState = { ...previousState };
+
+    if (type === 'addProperties') {
+      newState = Object.assign({}, previousState, action.extraData);
```

Comment:
[CODE STYLE] - Since you're already creating a new object with the spread operator in line 17, the use of `Object.assign` in line 20 is redundant. You can directly add properties to `newState` using the spread operator or by assigning them directly.